services/gateway/gateway-service.mjs

```diff
@@ -310,24 +346,22 @@ class GatewayService {
     return true;
   }
 
-  _getDistanceSquare(a,b){
-    return Math.pow(Math.abs(a.x - b.x),2) + Math.pow(Math.abs(a.y - b.y),2);
+  _getManhattanDistance(a,b){
+    return Math.abs(a.x - b.x) + Math.abs(a.y - b.y);
   }
 
-  _borderCheck(msg,mapSize){
-    if(msg.x > mapSize.width || msg.x < 0){
+  _borderCheck(coord,mapSize){
+    if(coord.x > mapSize.width || coord.x < 0){
```

Comment:
Add jsdoc, please.
The available range of x coordinate is [0, mapSize.width), so the check should be
```
if (coord.x >= mapSize.width || ...)
```

Reply:
I'll leave the task of adding jsdoc to whyang.